src/test/java/edu/indiana/dlib/amppd/service/BatchServiceTests.java

```diff
@@ -0,0 +1,341 @@
+        
+        Assert.assertTrue(response.hasErrors());
+	}
+	
+	/*
+	 * Invalid primary file with supplement file type = primary
+	 */
+	@Test
+	public void shouldBeInvalidPrimaryFileWithPrimarySupplement() throws Exception {
+		String fileName = "batch_manifest_invalid_primary_file_2.csv";
+		ClassLoader classLoader = ClassLoader.getSystemClassLoader();
+ 
+        File file = new File(classLoader.getResource(fileName).getFile());
+        String content = new String(Files.readAllBytes(file.toPath()));
+
+        Optional<AmpUser> users = ampUserRepository.findByUsername("dan");
+        
+        BatchValidationResponse response = manifestService.validate("Test Unit", "Test File", users.get(), content);
+        
+        Assert.assertTrue(response.hasErrors());
+	}
+	
+	/*
+	 * Invalid supplement file
+	 */
+	@Test
+	public void shouldBeInvalidSuppFile() throws Exception {
+		String fileName = "batch_manifest_invalid_supp_file.csv";
+		ClassLoader classLoader = ClassLoader.getSystemClassLoader();
+ 
+        File file = new File(classLoader.getResource(fileName).getFile());
+        String content = new String(Files.readAllBytes(file.toPath()));
+
+        Optional<AmpUser> users = ampUserRepository.findByUsername("dan");
+        
+        BatchValidationResponse response = manifestService.validate("Test Unit", "Test File", users.get(), content);
+        
+        Assert.assertTrue(response.hasErrors());
+	}
+	
+	/*
+	 * Helper method to unzip zip file to destination directory
+	 */
+	private void unzip(String zipFilePath, String destDirectory) throws IOException {
+        File destDir = new File(destDirectory);
+        if (!destDir.exists()) {
+            destDir.mkdir();
+        }
+        ZipInputStream zipIn = new ZipInputStream(new FileInputStream(zipFilePath));
+        ZipEntry entry = zipIn.getNextEntry();
+        // iterates over entries in the zip file
+        while (entry != null) {
+            String filePath = destDirectory + File.separator + entry.getName();
+            if (!entry.isDirectory()) {
+                // if the entry is a file, extracts it
+                extractFile(zipIn, filePath);
+            } else {
+                // if the entry is a directory, make the directory
+                File dir = new File(filePath);
+                dir.mkdir();
+            }
+            zipIn.closeEntry();
+            entry = zipIn.getNextEntry();
+        }
+        zipIn.close();
+    }
+	/*
+	 * Extracts individual file from zip
+	 */
+	private void extractFile(ZipInputStream zipIn, String filePath) throws IOException {
+        BufferedOutputStream bos = new BufferedOutputStream(new FileOutputStream(filePath));
+        byte[] bytesIn = new byte[4096];
+        int read = 0;
+        while ((read = zipIn.read(bytesIn)) != -1) {
+            bos.write(bytesIn, 0, read);
+        }
+        bos.close();
+    }
+	
+}
+
+
```

Comment:
Could you remove the .DS_Store below (maybe add to gitignore for the future)?

Reply:
Added it to gitignore and removed the file.